src/ape_ethereum/ecosystem.py

```diff
@@ -248,26 +248,22 @@ def decode_receipt(self, data: dict) -> ReceiptAPI:
         if txn_hash:
             txn_hash = data["hash"].hex() if isinstance(data["hash"], HexBytes) else data["hash"]
 
-        input_data = data.get("data") or data.get("input", b"")
-        if isinstance(input_data, str):
-            input_data = bytes(HexBytes(input_data))
+        if isinstance(data.get("data") or data.get("input", b""), str):
+            if data.get("data"):
+                data["data"] = bytes(HexBytes(data.get("data")))  # type: ignore
+            elif data.get("input", b""):
+                data["input"] = bytes(HexBytes(data.get("input", b"")))
```

Comment:
I think this can be simplified, I noticed in the old code that `input_data` doesn't seem to be used?

Reply:
input data was never used, so I removed it. I do not know why that was there originally, and I didn't want to just remove the conversion in case we had plans for it in the future.

As for simpler/cleaner. Maybe something like this:

```python
if data.get("data") and isinstance(data.get("data"), str):
    data["data"] = bytest(HexBytes(data.get("data")))

elif data.get("input") and isinstance(data.get("input", b""), str):
    data["input"] = bytes(HexBytes(data.get("input")))
```
If it goes to get "data" from the dictionary, and it does not exist, it never checks the instance. Not sure how to one-line this, and even if we did, I'm not sure it's cleaner

Reply:
[9b6cbf0](https://github.com/ApeWorX/ape/pull/1015/commits/9b6cbf058db760e44302ccb900e879443ed99bc9)

I think this was the commit?